crates/ruff_linter/resources/test/fixtures/ruff/for_variable_used_after_block.py

```diff
@@ -0,0 +1,81 @@
+for global_var in []:
+    _ = global_var
+    pass
+
+# ❌
+_ = global_var
+
+def foo():
+    # For control var used outside block
+    for event in []:
+        _ = event
+        pass
+
+    # ❌
+    _ = event
+
+    for x in []:
+        _ = x
+        pass
+    
+    # Using the same control variable name in a different loop is ok
+    for x in []:
+        _ = x
+        pass
+
+    for y in []:
+        # ❌ x is used outside of the loop it was defined in (meant to use y)
+        if x == 5:
+            pass
+
+    # Assign a variable before the loop
+    room_id = 3
+    _ = room_id
+    # Use the same variable name in a loop
+    for room_id in []:
+        _ = room_id
+        pass
+
+    # ❌ After the loop is not ok because the value is probably not what you expect
+    _ = room_id
+
+    # ❌ Augmented assignment is not allowed because the value is probably not what you expect
+    room_id += 1
+
+    # Assigning again after the loop is ok
+    room_id = 5
+    room_id += 1
+    _ = room_id
```

Comment:
So, I think there are two different reasons why you want to avoid using a loop variable after the loop has finished:
1. The first reason is that the loop might be empty, which means that the variable is never assigned:
   ```py
   iterable = []
   for x in iterable:
       pass
   print(x)  # NameError
   ```
   Catching these feels like a "correctness" issue, that we should hopefully be able to do without very many false positives.
2. The second is cases where we can be sure that the variable is defined, but we still forbid this pattern, perhaps on grounds of readability/style (it might not be obvious to the reader that the variable is mutated; there might be clearer ways of writing the code), or because there's a chance that the author of the code might not be familiar with Python's scoping rules, and might not have intended to reassign the variable by using the loop:
   ```py
   x = 42
   for x in range(5):
        pass
   print(x)  # oops, now it's 4?? The author of the code expected it to still be 42
   ```
   Sometimes code like this _can_ indicate a bug, but it also might be working exactly as the author of the code intended; there's a lot of code out there that deliberately mutates a variable using a loop like this.

The conclusion I draw from this is that we in fact need two rules here:
1. One rule to catch instances where a loop variable is used after the loop and was not defined before the loop. This is almost always bad practice, as the loop variable might not be defined.
2. One more opinionated rule that catches instances where the loop variable is definitely defined, but the mutation of the variable via the loop might not be desired. This might be a bug and might not be, depending on how experienced the developer is with Python's semantics.

For now I think I'd implement the first rule; we can consider the second rule as a followup.